src/buildCommon.js

```diff
@@ -334,31 +318,37 @@ const makeVList = function(children, positionType, positionData, options) {
         depth = 0;
     }
 
-    // Make the fontSizer
-    let maxFontSize = 0;
+    // Create a strut that is taller than any list item. The strut is added to
+    // each item, where it will determine the item's baseline. Since it has
+    // `overflow:hidden`, the strut's top edge will sit on the item's line box's
+    // top edge and the strut's bottom edge will sit on the item's baseline,
+    // with no additional line-height spacing. This allows the item baseline to
+    // be positioned precisely without worrying about font ascent and
+    // line-height.
+    let pstrutSize = 0;
     for (i = 0; i < children.length; i++) {
         if (children[i].type === "elem") {
-            maxFontSize = Math.max(maxFontSize, children[i].elem.maxFontSize);
+            const child = children[i].elem;
+            pstrutSize = Math.max(pstrutSize, child.maxFontSize, child.height);
         }
     }
-    const fontSizer = makeFontSizer(options, maxFontSize);
+    pstrutSize += 2;
+    const pstrut = makeSpan(["pstrut"], []);
+    pstrut.style.height = pstrutSize + "em";
 
     // Create a new list of actual children at the correct offsets
     const realChildren = [];
+    let minPos = depth;
+    let maxPos = 0;
     currPos = depth;
     for (i = 0; i < children.length; i++) {
         if (children[i].type === "kern") {
             currPos += children[i].size;
         } else {
             const child = children[i].elem;
 
-            const shift = -child.depth - currPos;
-            currPos += child.height + child.depth;
-
-            const childWrap = makeSpan([], [fontSizer, child]);
-            childWrap.height -= shift;
-            childWrap.depth += shift;
-            childWrap.style.top = shift + "em";
+            const childWrap = makeSpan([], [pstrut, child]);
+            childWrap.style.top = (-pstrutSize - currPos - child.depth) + "em";
```

Comment:
Just so I understand this correctly, the `pstrut` in the `childWrap` is taller than the `child` so that we avoid issues with the baseline alignment and can align the bottom of the child's glyph bounding box with the bottom of the `pstrut`?  And then `childWrap` is position vertically so that `child` appears where we want it to.